test/utils/ruleset-test-harness.ts

```diff
@@ -45,14 +45,14 @@ const nodeTestHarness: RulesetTestHarness = {
     return Promise.resolve();
   },
 
-  popEvent: async (timeoutMs: number = 1000) => new Promise<any[]>((resolve, reject) => {
+  popEvent: async (timeoutMs: number = 1000) => new Promise<any>((resolve) => {
     const startTime = new Date().getTime();
 
     const checkForEvents = setInterval(() => {
       const currentTime = new Date().getTime();
       if (currentTime - startTime >= timeoutMs) {
         clearInterval(checkForEvents);
-        reject(new Error(`Timeout ${timeoutMs} exceeded waiting for FS.event call.`));
+        resolve(undefined);
```

Comment:
Instead of rejecting the promise which throws when awaited, it seemed nicer to return `undefined` which could be asserted against for tests verifying no `FS.event` calls were queued.